transformation/plugins/org.jboss.tools.fuse.transformation.editor/src/main/java/org/jboss/tools/fuse/transformation/editor/internal/wizards/Hl7Page.java

```diff
@@ -0,0 +1,570 @@
+            @Override
+            public void widgetSelected(final SelectionEvent event) {
+                final String extension = "json";
+                final String path = selectResourceFromWorkspace(_page.getShell(), extension);
+                if (path != null) {
+                    try {
+                        hl7PreviewText.setText("");
+                        if (fileIsEmpty(path)) {
+                            hl7FileText.setText(path);
+                            notifyControl(hl7FileText, SWT.Modify);
+                            return;
+                        }
+                        IPath tempPath = new Path(path);
+                        IFile xmlFile = model.getProject().getFile(tempPath);
+                        String jsonText = getJsonText(xmlFile);
+                        if (!Util.isJSONValid(jsonText)) {
+                            hl7FileText.setText(path);
+                            notifyControl(hl7FileText, SWT.Modify);
+                            return;
+                        }
+
+                        boolean schema = jsonSchema(path);
+                        hl7InstanceOption.setSelection(!schema);
+                        hl7SchemaOption.setSelection(schema);
+                        if (isSourcePage()) {
+                            if (schema) {
+                                model.setSourceType(ModelType.JSON_SCHEMA);
+                            } else {
+                                model.setSourceType(ModelType.JSON);
+                            }
+                            model.setSourceFilePath(path);
+                        } else {
+                            if (schema) {
+                                model.setTargetType(ModelType.JSON_SCHEMA);
+                            } else {
+                                model.setTargetType(ModelType.JSON);
+                            }
+                            model.setTargetFilePath(path);
+                        }
+                        hl7FileText.setText(path);
+                        hl7PreviewText.setText(jsonText);
+                        notifyControl(hl7FileText, SWT.Modify);
+                    } catch (final Exception e) {
+                        Activator.error(e);
+                    }
+                }
+            }
+        });
+
+        final Group group2 = new Group(_page, SWT.SHADOW_ETCHED_IN);
+        group2.setText("JSON Structure Preview");
+        group2.setLayout(new FillLayout());
+        group2.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true, 3, 3));
+
+        hl7PreviewText = new Text(group2, SWT.V_SCROLL | SWT.READ_ONLY | SWT.H_SCROLL );
+        hl7PreviewText.setBackground(_page.getBackground());
+
+        bindControls();
+        validatePage();
+    }
+
+    private String getJsonText(IFile jsonFile) {
+        if (jsonFile != null && jsonFile.exists()) {
+            try (final InputStream istream = jsonFile.getContents()) {
```

Comment:
update preview and getJSonText seems to have some duplkcated lines of code.

Btw, with Java 7, it is now simpler, new String (Files.readAllBytes(file.toPath()),Charset.forName("UTF-8"))
see http://www.adam-bien.com/roller/abien/entry/java_8_reading_a_file